### ISSUE TITLE
Data type "Array" in "Podcasts.tags" is not supported by "postgres" database.

### ISSUE THREAD

**Issue by isiah-lloyd** [typeorm/typeorm#2760]
**Issue type:**

[x] question
[ ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x ] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I have an entity with the following a column set as an array and I'm using postgres but I'm getting an error: 
`DataTypeNotSupportedError: Data type "Array" in "Podcasts.tags" is not supported by "postgres" database.` 

This is the snippet from the entity that is relevant:
```
  @Column({array: true})
  tags: string[];
```

**Comment by isiah-lloyd** [comment]
This was a simple fix, the type of the column shouldn't be an array. Changed to this and it now works: 
```
  @Column({array: true})
  tags: string[];
```

Better documentation on arrays would have been useful.

**Comment by dtruel** [comment]
I have this problem too, and I don't think that is the right answer because then how are you supposed to access the array elements?  According to this page, you should be able to put an array

**Comment by isiah-lloyd** [comment]
I haven't tested my solution yet only that it compiled without errors so I'm going to reopen this and see if we can figure this out.

**Comment by dtruel** [comment]
Good idea. I have the exact same problem. Only switching to `string` doesn't even fix it for me.  I'm trying to do the exact same thing as you - add a "tags" array.

**Comment by dtruel** [comment]
What about this:

```typescript
 @Column("text", {array: true})
  tags: string;
```

**Comment by wyqydsyq** [comment]
I've tried following the suggestion from @turbobuilt but I now get:
```
could not find array type for data type text[]
```
When trying to query this column, any ideas? Is there some extra setup I need to do to get TypeORM to create the appropriate data types? I'm already starting it with `sync: true`

**Comment by TharakaSl** [comment]
@wyqydsyq Did you fix the 'could not find array type for data type text[]' error?

**Comment by felinto-dev** [comment]
> ```ts
>  @Column("text", {array: true})
>   tags: string[];
> ```

You're a genius! Thanks so much!

**Comment by mtenorio-cmp** [comment]
en MySql

```
 @Column({ type: 'simple-array' })
  tags: string[];
```

**Comment by adarshaacharya** [comment]
What's the difference between `array` and `simple-array` in typeorm ?

eg : 
array of string:
```
  @Column('simple-array')
  tags : string[]
```
simple array : 
```
  @Column('string')
  tags : string[]
```

**Comment by zhangciwu** [comment]
Found this works if you don't want `text`:
```
@Column('varchar', { array: true })
  tags: string[];
```

**Comment by Merdi-kim** [comment]
I'm adding this just in case someone falls into the same situation. 
Here are 2 quick ways to fix the problem.

- Replace @column( 'string',  { array : true } ) by @column( 'text',  { array : true } )
- Replace @column( 'string',  { array : true } ) by @column( 'varchar',  { array : true } )

then add the column name and define the type as follow: column_name : string[ ]

**Comment by shabbyaaa** [comment]
if i have two tables, andthey have many-to-many relationships, how do i define this structure, when i use this approach, i get errors like this:
```
  @Column({ comment: 'tag', array: true })
  @ManyToMany(() => Tag, (tag) => tag.tagId)
  @JoinTable({
    name: 'article_tag',
    joinColumns: [
      {
        name: 'article_id',
      },
    ],
    inverseJoinColumns: [
      {
        name: 'tag_id',
      },
    ],
  })
  tags: Tag[];

DataTypeNotSupportedError: Data type "Array" in "Article.tags" is not supported by "postgres" database.

```
i want to konw how to use ManyToMany to define this structure?
